test/jdk/jdk/nio/zipfs/ZipFSTester.java

```diff
@@ -686,10 +686,11 @@ static void test8131067() throws Exception {
     private static FileSystem newZipFileSystem(Path path, Map<String, ?> env)
         throws Exception
     {
+        var plusReplacedUri = path.toUri().toString().replace("+", "%2b");
```

Comment:
Thank you for your proposed patch.

The path described in the bug  is a bit unusual and not something we typically expect (or have seen).

Please add a comment describing the purpose(maybe include a trivial example) of this change to make it more readily apparent to future maintainers.

Not sure I am enamored with the name of the variable `plusReplacedUri`.  Perhaps consider an alternative name.

Reply:
Thanks. It is a good idea to comment more information for maintainers.

BTW, this bug is caused by the `newZipFileSystem` method, which assert its parameter `Path path` is utf-8 encoded.
However, the path would not be encoded in most cases.
So the `URLDecoder.decode(path.toUri().toString(), "utf8")` at line 692 may cause unexpected results, 

Maybe just remove the utf-8 decode logic is a valid option. What do you think?

Reply:
We should leave it as is for now as it does test newFileSystem via a URI which is also supported.

Certainly it could be updated at a later date when replacement is added(or at least more coverage) to exercise Zip FS, newFileSystem via URI.